
From: Simon Derr <Simon.Derr@bull.net>

This patch fixes a trivial bug, triggered when using the cpusets as a
non-root user.

Signed-off-by: Paul Jackson <pj@sgi.com>
Signed-off-by: Simon Derr <simon.derr@bull.net>
Signed-off-by: Andrew Morton <akpm@osdl.org>
---

 25-akpm/kernel/cpuset.c |    2 +-
 1 files changed, 1 insertion(+), 1 deletion(-)

diff -puN kernel/cpuset.c~cpusets-fix-cpuset_get_dentry kernel/cpuset.c
--- 25/kernel/cpuset.c~cpusets-fix-cpuset_get_dentry	2004-10-05 11:14:35.534316632 -0700
+++ 25-akpm/kernel/cpuset.c	2004-10-05 11:14:35.538316024 -0700
@@ -235,7 +235,7 @@ static struct dentry *cpuset_get_dentry(
 	qstr.len = strlen(name);
 	qstr.hash = full_name_hash(name, qstr.len);
 	d = lookup_hash(&qstr, parent);
-	if (d)
+	if (!IS_ERR(d))
 		d->d_op = &cpuset_dops;
 	return d;
 }
_
